mastering/build_files.py

```diff
@@ -18,3 +37,348 @@ def getFiles(path, extension):
+                     decimals=1, length=100, fill='█'):
+    """
+    Call in a loop to create terminal progress bar
+    @params:
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : positive number of decimals in percent complete (Int)
+        length      - Optional  : character length of bar (Int)
+        fill        - Optional  : bar fill character (Str)
+
+    Nabbed, of course, from Stack Overflow
+    (https://stackoverflow.com/questions/3173320/text-progress-bar-in-the-console)
+    """
+    percent = ("{0:."+str(decimals)+"f}").format(100*(iteration/float(total)))
+    filledLength = int(length * iteration // total)
+    bar = fill * filledLength + '-' * (length - filledLength)
+    print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end='\r')
+    # Print New Line on Complete
+    if iteration == total:
+        print()
+
+
+def buildStatic(designspacePath, outputPath):
+    """
+    Reads designspace file, generates instance UFOs,
+    builds mastering files for the Adobe FDK, then
+    builds files.
+
+    *designspacePath* is the path to the designspace file.
+    *outputPath* is the path for file builds.
+
+    """
+    pass
+
+
+def buildFontMenuDB(designspace, root):
+    """
+    Pulls the postscript, and the mapping names from the designspace.
+    Transforms the family name from "Recursive" to either "Recursive Sans"
+    or "Recursive Mono" for the static fonts.
+    """
+
+    out = ""
+
+    for i in designspace.instances:
+        fn = i.familyName + " " + i.styleName.split()[0]
+        sn = " ".join(i.styleName.split()[1:])
+        out += f"[{i.postScriptFontName}]\n"
+        out += f"    f={fn}\n"
+        out += f"    s={sn}\n"
+        out += f"    l={i.styleMapFamilyName}\n"
+        out += f"    m=1,{i.styleMapFamilyName} {i.styleMapStyleName}\n\n"
+
+    path = os.path.join(root, "FontMenuNameDB")
+    with open(path, "w") as f:
+        f.write(out)
+
+
+def buildGlyphOrderAndAlias(designspace):
+    pass
+
+
+def buildFontInfo(font):
+    pass
+
+
+def buildFeatures(font):
+    pass
+
+
+def fillInPanoseValues(font):
+    """
+    Fills in the Panose values for a font based on the style name.
+    Uses the  panoseWeightMap to set weight values, the presense of
+    'Mono' to set if it's monospaced, and the presense of 'Italic' to
+    set if it's oblique. 'Casual' and 'Linear' trigger settings for
+    the style.
+    """
+
+    panoseWeightMap = {
+                       "Black": 11,
+                       "UltraBold": 10,
+                       "ExtraBold": 9,
+                       "Bold": 8,
+                       "SemiBold": 7,
+                       "Medium": 6,
+                       "Regular": 5,
+                       "Light": 4,
+                      }
```

Comment:
We should also update these instance names, correct?

Reply:
Fixed